ui/src/components/EntityModal/EntityModal.test.tsx

```diff
@@ -340,3 +343,106 @@ describe('Default value', () => {
+        }
+        return [endpointAuth, endpointToken];
+    };
+
+    const spyOnWindowOpen = async (addButton: HTMLElement) => {
+        const windowOpenSpy = jest.spyOn(window, 'open') as jest.Mock;
+
+        // mock opening verification window
+        windowOpenSpy.mockImplementation((url) => {
+            expect(url).toEqual(
+                'https://authendpoint/services/oauth2/authorize?response_type=code&client_id=Client%20Id&redirect_uri=http%3A%2F%2Flocalhost%2F'
+            );
+
+            return { closed: true };
+        });
+
+        await userEvent.click(addButton);
+        windowOpenSpy.mockRestore();
+    };
+
+    const props = {
+        serviceName: 'account',
+        mode: 'create',
+        stanzaName: undefined,
+        formLabel: 'formLabel',
+        page: 'configuration',
+        groupName: '',
+        open: true,
+        handleRequestClose: () => {},
+    } satisfies EntityModalProps;
+
+    it('render modal with separated oauth fields', async () => {
+        setUpConfigWithSeparatedEndpoints();
+        renderModalWithProps(props);
+
+        const [endpointAuth, endpointToken] = await getFilledOauthFields();
+
+        expect(endpointAuth).toBeInTheDocument();
+        expect(endpointAuth).toHaveValue('authendpoint');
+        expect(endpointToken).toBeInTheDocument();
+        expect(endpointToken).toHaveValue('tokenendpoint');
+    });
+
+    it('check if correct authorization endpoint called', async () => {
+        setUpConfigWithSeparatedEndpoints();
+        renderModalWithProps(props);
+
+        await getFilledOauthFields();
+
+        const addButton = screen.getByRole('button', { name: /add/i });
+        expect(addButton).toBeInTheDocument();
+
+        await spyOnWindowOpen(addButton);
+    });
+
+    it('check if correct auth token endpoint created', async () => {
+        setUpConfigWithSeparatedEndpoints();
+        renderModalWithProps(props);
+        const backendTokenFunction = jest.fn();
+
+        await getFilledOauthFields();
+        const addButton = screen.getByRole('button', { name: /add/i });
+        expect(addButton).toBeInTheDocument();
+
+        await spyOnWindowOpen(addButton);
+
+        // token is aquired on backend side so only thing we can check is if there is correct url created
+        jest.spyOn(axiosWrapper, 'axiosCallWrapper').mockImplementation((params) => {
+            backendTokenFunction((params?.body as unknown as URLSearchParams)?.get('url'));
+            return new Promise((r) => r({} as unknown as PromiseLike<AxiosResponse>));
+        });
+
+        // triggering manually external oauth window behaviour after success authorization
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        (window as any).getMessage({ code: 200, msg: 'testing message for oauth' });
+
+        // only purpose is to check if backend function receinved correct token url
+        expect(backendTokenFunction).toHaveBeenCalledWith(
```

Comment:
please avoid testing implementation details such as a network calls.
Instead, emulate the response via msw and make assertions regarding the response of our component. 
If the auth response is successful, what should we see in the dom?

This is a great article https://kentcdodds.com/blog/testing-implementation-details

Reply:
hmmm generally it makes sense but for this case i'm not that convinced, 
i mean yea it is kind of testing implementation, but i'm here actually testing that specific thing, so if the correct url has been passed to backend.

To make it look a bit better I tried to use something like this 
```
    http.post(
        '/servicesNS/nobody/-/demo_addon_for_splunk_oauth/oauth',
        ({request}) => {
            // reading request request, displays object without specified data param, which should contain token url
            return HttpResponse.json({
                entry: [
                    {
                        content: {
                            accessToken: 'accessToken1111111',
                            instanceUrl: 'instanceUrl22222222',
                            refreshToken: 'refreshToken3333333',
                        },
                    },
                ],
            });
        }
    ),
```
but for some reason `request` is empty and does not contain data parameter, do you know what might be a cause of it, tried implementations things without any luck?


Also if that change would be hard for now i would leave it as it is, or just remove that test completely, as whole purpose of it is to verify if correct url is passed to backend, but do not verify its response or whatever is happening latter with data and flow, that should be solved with another test, to keep it separated from this one

Reply:
When I was reviewing the PR put a comment but puzzled how would I make it. And I tried and fail to avoid implementation details. 
And I removed my comment (I thought), sorry for the confusion.